assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/segment_anything/segment_anything_mlflow_wrapper.py

```diff
@@ -0,0 +1,178 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""
+Contains MLFlow pyfunc wrapper for segment anything models.
+
+Has methods to load the model and predict.
+"""
+
+
+import io
+import mlflow
+import pandas as pd
+from PIL import Image
+import torch
+from transformers import SamModel, SamProcessor
+
+from config import MLflowSchemaLiterals, Tasks, MLflowLiterals, SAMHFLiterals, DatatypeLiterals
+
+
+class SegmentAnythingDiffusionMLflowWrapper(mlflow.pyfunc.PythonModel):
+    """MLflow model wrapper for segment anything models."""
+
+    def __init__(
+        self,
+        task_type: str,
+    ) -> None:
+        """Initialize model parameters for converting Huggingface segment anything model to mlflow.
+
+        :param task_type: Task type used in training.
+        :type task_type: str
+        """
+        super().__init__()
+        self._task_type = task_type
+
+    def load_context(self, context: mlflow.pyfunc.PythonModelContext) -> None:
+        """
+        Load a MLflow model with pyfunc.load_model().
+
+        :param context: MLflow context containing artifacts that the model can use for inference
+        :type context: mlflow.pyfunc.PythonModelContext
+        """
+        if self._task_type == Tasks.SEGMENT_ANYTHING.value:
+            try:
+                _map_location = "cuda" if torch.cuda.is_available() else "cpu"
+                model_dir = context.artifacts[MLflowLiterals.MODEL_DIR]
+                self._model = SamModel.from_pretrained(model_dir).to(_map_location)
+                self._processor = SamProcessor.from_pretrained(model_dir)
+                print("Model loaded successfully")
+            except Exception as e:
+                print("Failed to load the the model.")
+                print(e)
+                raise
+        else:
+            raise ValueError(f"invalid task type {self._task_type}")
+
+    def predict(
+        self,
+        context: mlflow.pyfunc.PythonModelContext,
+        input_data: pd.DataFrame,
+    ) -> pd.DataFrame:
+        """
+        Perform inference on the input data.
+
+        :param context: MLflow context containing artifacts that the model can use for inference
+        :type context: mlflow.pyfunc.PythonModelContext
+        :param input_data:
+            Pandas DataFrame with a first column name ["image"] which is in base64 String format or an URL, and
```

Comment:
nit: the desc for input_data has weird \n(s).

Reply:
Didn't get it, can you open it via view file option and point out the area?

Reply:
![image](https://github.com/Azure/azureml-assets/assets/114467536/5f55562c-fd07-402c-a4b3-8a873e408ce0)

Reply:
Copied from official docstring: https://github.com/huggingface/transformers/blob/574a5384557b1aaf98ddb13ea9eb0a0ee8ff2cb2/src/transformers/models/sam/modeling_sam.py#L1132